src/transformers/models/ernie_m/tokenization_ernie_m.py

```diff
@@ -0,0 +1,356 @@
+            A special token used for sequence classification. It is the last token of the sequence when built with
+            special tokens. Defaults to "[CLS]".
+        mask_token (str, optional):
+            A special token representing a masked token. This is the token used in the masked language modeling task
+            which the model tries to predict the original unmasked ones. Defaults to "[MASK]".
+    """
+    resource_files_names = {
+        "sentencepiece_model_file": "sentencepiece.bpe.model",
+        "vocab_file": "vocab.txt",
+    }  # for save_pretrained
+    pretrained_resource_files_map = {
+        "vocab_file": {
+            "ernie-m-base": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/vocab.txt",
+            "ernie-m-large": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/vocab.txt",
+        },
+        "sentencepiece_model_file": {
+            "ernie-m-base": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/sentencepiece.bpe.model",
+            "ernie-m-large": "https://huggingface.co/susnato/ernie-m-base_pytorch/blob/main/sentencepiece.bpe.model",
+        },
+    }
+    pretrained_init_configuration = {
+        "ernie-m-base": {"do_lower_case": False},
+        "ernie-m-large": {"do_lower_case": False},
+        "uie-m-base": {"do_lower_case": False},
+        "uie-m-large": {"do_lower_case": False},
+    }
+    max_model_input_sizes = {"ernie-m-base": 514, "ernie-m-large": 514, "uie-m-base": 514, "uie-m-large": 514}
+    # Ernie-M model doesn't have token_type embedding.
+    model_input_names: List[str] = ["input_ids"]
+
+    vocab_files_names = VOCAB_FILES_NAMES
+
+    def __init__(
+        self,
+        vocab_file,
+        sentencepiece_model_ckpt,
+        do_lower_case=False,
+        encoding="utf8",
+        unk_token="[UNK]",
+        sep_token="[SEP]",
+        pad_token="[PAD]",
+        cls_token="[CLS]",
+        mask_token="[MASK]",
+        **kwargs
+    ) -> None:
+        # Mask token behave like a normal word, i.e. include the space before it and
+        # is included in the raw text, there should be a match in a non-normalized sentence.
+
+        self.sentencepiece_model_ckpt = {} if sentencepiece_model_ckpt is None else sentencepiece_model_ckpt
+
+        super().__init__(
+            do_lower_case=do_lower_case,
+            unk_token=unk_token,
+            sep_token=sep_token,
+            pad_token=pad_token,
+            cls_token=cls_token,
+            mask_token=mask_token,
+            sp_model_kwargs=self.sentencepiece_model_ckpt,
+            **kwargs,
+        )
+
+        self.do_lower_case = do_lower_case
+        self.sp_model = spm.SentencePieceProcessor()
+        self.sp_model.Load(sentencepiece_model_ckpt)
+        self.vocab = self.load_vocab(filepath=vocab_file)
+        self.reverse_vocab = dict((v, k) for k, v in self.vocab.items())
+
+        assert len(self.vocab) == len(self.reverse_vocab)
```

Comment:
avoid using `assert` please, test the condition and raise a relevant error instead ;)